src/main/com/example009/radek/v3/FileService.java

```diff
@@ -0,0 +1,133 @@
+package main.com.example009.radek.v3;
+
+import java.io.*;
+import java.util.List;
+import java.util.Scanner;
+
+public class FileService {
+
+    private final String PATH_FILE_RESULTS = "src/main/resources/example009/v3/";
+
+    void storeResultsIntoFile(List<Double> inputList) {
+
+        String name = getUserInput("Please insert file name where you want to save results: ");
+        saveResultsIntoFile(name, inputList);
+    }
+
+    void loadResultsFromFile(String name) throws IOException {
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(PATH_FILE_RESULTS + name))) {
+            reader.lines().forEach(System.out::println);
+        } catch (FileNotFoundException e) {
+            throw new FileNotFoundException("Connecting to file reader wasn't successful");
+        } catch (IOException e) {
+            throw new IOException("Reading from file wasn't successful");
+        }
+    }
+
+    String getUserInput(String message) {
+
+        System.out.println(message);
+        Scanner scanner = new Scanner(System.in);
+        String input = scanner.nextLine();
+        if (input.isEmpty() || input.equals("exit")) {
+            System.out.println("Type valid response, or type 'exit' to stop the program");
+            input = getUserInput();
+        }
+        return input;
+    }
+
+    int getUserIntInput() {
+
+        System.out.println("Please insert file number or type '0' to end program");
+        Scanner scanner = new Scanner(System.in);
+        int input = scanner.nextInt();
+        if (input == 0) {
+            System.exit(0);
+        }
+        return input;
+    }
+
+    String getCheckedUserInput() {
+
+        System.out.println("Please insert file name where you want to load results from: ");
+        Scanner scanner = new Scanner(System.in);
+        String input = scanner.nextLine();
+        boolean exists = fileExists(input);
+        while (!exists || input.equals("exit")) {
+            System.out.println("File name isn't correct try again, or type 'exit' to stop the program");
+            input = getUserInput();
+            exists = fileExists(input);
+        }
+        if (input.isEmpty()) {
+            System.out.println("Type valid file name, or type 'exit' to stop the program");
+            input = getUserInput();
+        }
+        return input;
+    }
+
+    String getUserInput() {
+
+        String input;
+        do {
+            Scanner scanner = new Scanner(System.in);
+            input = scanner.nextLine();
+            if (input.equals("exit")) {
+                System.exit(0);
+            }
+        } while (input.isEmpty());
+
+        return input;
+    }
+
+    void loadPreviousResultFile() throws IOException {
+
+        File file = new File(PATH_FILE_RESULTS);
+        String[] files = file.list();
+        if (files != null && files.length > 0) {
+            if (getUserInput("Do you want to load previous results ? yes/no").equals("yes")) {
```

Comment:
if ... ptas se jestli je neco pravda ... musi to by strucne, rozhodne tam necpat string a jeste za to equals - neprijatelne

mas dva ify za sebou, to muze byt v jendom ifu -> pokud jsou podminky dlouhe a necitelne, lze zkratit pomoci dalsich metod: treba existAnyFile(files) && shouldLoadResultsFromFile()